soroban-sdk-auth/src/lib.rs

```diff
@@ -0,0 +1,112 @@
+#![no_std]
+
+use soroban_sdk::{serde::Serialize, Account, BigInt, BytesN, Env, EnvVal, Symbol, TryIntoVal};
+
+pub mod public_types;
+use crate::public_types::{
+    Identifier, KeyedAccountAuthorization, KeyedAuthorization, KeyedEd25519Signature, Message,
+    MessageV0,
+};
+
+pub trait NonceAuth {
+    fn read_nonce(e: &Env, id: Identifier) -> BigInt;
+    fn read_and_increment_nonce(&self, e: &Env, id: Identifier) -> BigInt;
+    fn get_keyed_auth(&self) -> &KeyedAuthorization;
+}
+
+pub fn check_ed25519_auth(
+    e: &Env,
+    auth: &KeyedEd25519Signature,
+    function: Symbol,
+    parameters: EnvVal,
+) {
+    let msg = MessageV0 {
+        function,
+        contrct_id: e.get_current_contract(),
+        network_id: e.ledger().network_passphrase(),
+        parameters: parameters.to_raw().try_into_val(e).unwrap(),
```

Comment:
These should probably be called args rather than parameters since they're the actual values, not the parameter list.
```suggestion
        args: args.to_raw().try_into_val(e).unwrap(),
```

Reply:
Done